### PR TITLE
docs(migration): add missing Collector and Local File Source details

### PR DESCRIPTION
**Pull request by sumo-drosiek**

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -363,6 +364,22 @@ receivers:
     multiline:
       ## line_start_pattern is substitute of `Boundary Regex`.
       line_start_pattern: ^\d{4}
+    ## Adds file path log.file.path attribute, which can be used for timestamp parsing
+    ## See operators configuration
+    include_file_path: true
+    ## `Operators allows to perform more advanced operations like per file timestamp parsing
+    operators:
+    - type: regex_parser
+      ## Applies only to tmp/logs.log file
```
Comment (sumo-drosiek) [inline docs/migration.md:373]:
```suggestion
      ## Apply only to tmp/logs.log file
```

File: docs/migration.md [hunk 2]
```diff
@@ -363,6 +364,22 @@ receivers:
...
+      if: 'attributes["log.file.path"] == "tmp/logs.log"'
+      ## Extracts timestamp to timestamp_field using regex parser
+      regex: '^(?P<timestamp_field>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}).*'
+      timestamp:
+        parse_from: attributes.timestamp_field
+        layout: '%Y-%m-%d %H:%M:%S'
+    ## cleanup timestamp_field
```
Comment (sumo-drosiek) [inline docs/migration.md:380]:
```suggestion
    ## Cleanup timestamp_field
```

File: docs/migration.md [hunk 3]
```diff
@@ -371,15 +388,29 @@ processors:
     source_category: example category
     ## Installed Collector substitute for `Source Host`.
     source_host: example host
-exporters:
-  sumologic:
-    ## clear_logs_timestamp is by default set to True.
-    ## If it's set to true, it works like `Enable Timestamp Parsing`,
-    ## and `Time Zone` is going to be taken from `extensions` section.
+  transform/clear_logs_timestamp:
+    ## By default every data has timestamp (usually set to receive time)
```
Comment (sumo-drosiek) [inline docs/migration.md:392]:
```suggestion
    ## By default every data has timestamp (usually set to receipt time)
```

File: docs/migration.md [hunk 4]
```diff
@@ -371,15 +388,36 @@ processors:
...
 exporters:
   sumologic:
-    ## clear_logs_timestamp is by default set to True.
-    ## If it's set to true, it works like `Enable Timestamp Parsing`,
-    ## and `Time Zone` is going to be taken from `extensions` section.
-    ## There is no possibility to configure several time zones in one exporter.
-    ## clear_logs_timestamp sets to true also behaves like
-    ## `Timestamp Format` would be set to `Automatically detect the format`
-    ## in terms of Installed Collector configuration.
     clear_logs_timestamp: true
```
Comment (swiatekm-sumo) [inline docs/migration.md:421]: Shouldn't we delete this config value as well? In fact, I think we should delete the whole exporter section.

File: docs/migration.md [hunk 5]
```diff
@@ -713,8 +766,6 @@ processors:
     source_category: My Category
 exporters:
   sumologic/some_name:
-    ## Keep manually parsed timestamps
```
Comment (swiatekm-sumo) [inline docs/migration.md:716]: I think we can delete this as well.

File: docs/migration.md [hunk 6]
```diff
@@ -748,13 +799,11 @@ processors:
     source_category: My Category
 exporters:
   sumologic/some_name:
-    ## Keep manually parsed timestamps (use Receipt Time by default)
-    clear_logs_timestamp: true
```
Comment (swiatekm-sumo) [inline docs/migration.md:752]: And this.

File: docs/migration.md [hunk 7]
```diff
@@ -2055,39 +2104,39 @@ This section describes migration steps for an Installed Collector managed with a
...
+| `fipsJce=true/false`                          | N/A                                                                                       |
+| `hostName=hostname`                           | [processors.source.source_host][source-templates]                                         |
+| `name=name`                                   | [extensions.sumologic.collector_name](#name)                                              |
+| `proxyHost=host`                              | [plese see OTC documentation][proxy]                                                      |
+| `proxyNtlmDomain=NTLM domain`                 | [plese see OTC documentation][proxy]                                                      |
+| `proxyPassword=password`                      | [plese see OTC documentation][proxy]                                                      |
+| `proxyPort=port`                              | [plese see OTC documentation][proxy]                                                      |
+| `proxyUser=username`                          | [plese see OTC documentation][proxy]                                                      |
```
Comment (swiatekm-sumo) [inline docs/migration.md:2129]: `plese` -> `please`

File: docs/migration.md (hunk 4 above)
Comment (sumo-drosiek) [reply docs/migration.md:421]: We cannot delete the whole exporter section, as

```
Error: invalid configuration: no exporter configuration specified in config
2023/03/09 08:24:24 collector server run finished with error: invalid configuration: no exporter configuration specified in config
```

File: docs/migration.md (hunk 7 above)
Comment (sumo-drosiek) [reply docs/migration.md]: fixed

File: docs/migration.md (hunk 4 above)
Comment (swiatekm-sumo) [reply docs/migration.md]: My bad, from context I thought these were just snippets and not full configurations.

**Review by swiatekm-sumo — APPROVED** [review]

File: docs/migration.md (hunk 4 above)
Comment (sumo-drosiek) [reply docs/migration.md]: Maybe this should be changed. For now it's building configuration step by step

**Pull request merged**